spec/support/request/web_helper.rb

```diff
@@ -145,6 +170,28 @@ def select_select2_result_async(value)
     page.execute_script(%{$("div.select2-result-label:contains('#{value}')").mouseup()})
   end
 
+  def find_label_by_text(text)
+    label = find_label(text)
+    counter = 0
+
+    # Because JavaScript testing is prone to errors...
+    while label.nil? && counter < 10
+      sleep(1)
+      counter += 1
+      label = find_label(text)
+    end
+
+    if label.nil?
+      raise "Could not find label by text #{text}"
+    end
+
+    label
+  end
+
+  def find_label(text)
+    first(:xpath, "//label[text()[contains(.,'#{text}')]]")
```

Comment:
If we used `find` instead of `first` I bet the while loop above won't be needed. `find` is one of those Capybara waiting methods.

Reply:
I had a second look, I managed to delete this code, have a look at the new commit.